packages/web-components/src/components/masthead/masthead-composite.ts

```diff
@@ -967,24 +981,52 @@ class DDSMastheadComposite extends HostListenerMixin(LitElement) {
   }
 
   /**
-   * Handles the rendering of the megamenu once it is active
+   * Sets the active megamenu upon user interaction.
    *
    * @param event The event.
    */
-  @HostListener('eventMegamenuActive')
+  @HostListener(DDSMegamenuTopNavMenu.eventMegaMenuToggled)
   // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
   protected _loadMegamenu = (event: CustomEvent) => {
     const {
       target,
       detail: { active, resolveFn },
     } = event;
     const { autoid } = (target as HTMLElement).dataset;
-    const index = autoid?.slice(-1);
-    const currentMenu = this.megamenuSet[index!];
-    render(active ? currentMenu : nothing, target as HTMLElement);
+    const targetMegamenuIndex = Number(autoid?.slice(-1));
```

Comment:
Not a regression, but just noticing the `autoid?.slice(-1)` is fragile for the case when there are more than 9 items. Lets say there were 11, the first item and 11th item would both produce a `targetMegamenuIndex` of `1`. Very edge case, and not necessary to handle here, but I couldn't unsee it.